### PR TITLE
All tx sizes

### PR DESCRIPTION
**Pull request by tdaede**

### CODE REVIEW

File: src/quantize.rs [hunk 1]
```diff
@@ -15,9 +17,13 @@ pub fn ac_q(qindex: usize) -> i16 {
     }
 }
 
-pub fn quantize_in_place(qindex: usize, coeffs: &mut [i32]) {
-    coeffs[0] /= dc_q(qindex) as i32;
-    let ac_quant = ac_q(qindex) as i32;
+pub fn quantize_in_place(qindex: usize, coeffs: &mut [i32], tx_size: TxSize) {
+    let tx_scale = match tx_size {
```
Comment (ycho) [inline src/quantize.rs]: Nice to take care of this easy-to-make-mistake part!

File: src/quantize.rs [hunk 2]
```diff
@@ -26,11 +32,15 @@ pub fn quantize_in_place(qindex: usize, coeffs: &mut [i32]) {
     *coeffs.last_mut().unwrap() = 0;
 }
 
-pub fn dequantize(qindex:usize, coeffs: &[i32], rcoeffs: &mut [i32]) {
-    rcoeffs[0] = coeffs[0] * dc_q(qindex) as i32;
+pub fn dequantize(qindex:usize, coeffs: &[i32], rcoeffs: &mut [i32], tx_size: TxSize) {
+    let tx_scale = match tx_size {
```
Comment (ycho) [inline src/quantize.rs]: Nice here as well!

File: src/quantize.rs [hunk 3]
```diff
@@ -15,9 +17,13 @@ pub fn ac_q(qindex: usize) -> i16 {
...
 
-pub fn quantize_in_place(qindex: usize, coeffs: &mut [i32]) {
-    coeffs[0] /= dc_q(qindex) as i32;
-    let ac_quant = ac_q(qindex) as i32;
+pub fn quantize_in_place(qindex: usize, coeffs: &mut [i32], tx_size: TxSize) {
+    let tx_scale = match tx_size {
+        TxSize::TX_32X32 => 2,
+        _ => 1
+    };
+    coeffs[0] /= (dc_q(qindex) / tx_scale) as i32;
```
Comment (ycho) [inline src/quantize.rs]: Well, seeing again, 
doing coeffs[0] *= tx_scale; 
coeffs[0] /= dc_q(qindex) as i32;
does not lose the precision of dc_q(qindex)?

File: src/context.rs [hunk 4]
```diff
@@ -417,7 +417,7 @@ pub struct SCAN_ORDER {
 
   pub scan: &'static [u16; 64*64],
   pub iscan: &'static [u16; 64*64],
-  pub neighbors: &'static [u16; (64+1)*2]
+  pub neighbors: &'static [u16; (1024+1)*2]
```
[the remark this answers is not in the document]
Comment (tdaede) [reply src/context.rs]: Yeah it should be (64*64+1)*2. I'll be consistent.

File: src/quantize.rs (hunk 3 above)
Comment (tdaede) [reply src/quantize.rs]: Yeah I'll change it to do that.

**Comment by ycho** [comment]
Ok- looks good, and I've also tested again for speed=1,2,3,10 and worked fine. Pls merge.

**Pull request merged**